Python/optimizer.c

```diff
@@ -1659,4 +1668,61 @@ _Py_Executors_InvalidateAll(PyInterpreterState *interp, int is_invalidation)
     }
 }
 
+void _Py_Executor_Invalidate(_PyExecutorObject *executor)
+{
+   if (executor->vm_data.valid) {
+    unlink_executor(executor);
+    executor_clear(executor);
+   }
+}
+
+void
+_Py_Executors_InvalidateOld(PyInterpreterState *interp, int is_invalidation)
+{
```

Comment:
So, most of this body was copy-pasted from an earlier version of `_Py_Executors_InvalidateDependency`. It's since been cleaned up a bit... maybe re-copy-and-paste the implementation to use the simpler version? Just make sure to keep your changes.

I'm wondering if there's a way the code could be shared for both somehow, but that might be more trouble than it's worth...